target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-whw03v2.dts

```diff
@@ -0,0 +1,555 @@
+        };
+    };
+
+    serial_1_pins: serial1_pinmux {
+        mux {
+            pins = "gpio8", "gpio9", "gpio10", "gpio11";
+            function = "blsp_uart1";
+            bias-disable;
+        };
+    };
+
+    spi_0_pins: spi_0_pinmux {
+        pinmux {
+            function = "blsp_spi0";
+            pins = "gpio13", "gpio14", "gpio15";
+        };
+
+        pinmux_cs {
+            function = "gpio";
+			pins = "gpio12";
+        };
+
+		pinconf {
+			pins = "gpio13", "gpio14", "gpio15";
+			drive-strength = <12>;
+			bias-disable;
+		};
+
+        pinconf_cs {
+            pins = "gpio12";
+            drive-strength = <2>;
+            bias-disable;
+            output-high;
+        };
+    };
+    
+    spi_1_pins: spi_1_pinmux {
+        cs {
+            pins = "gpio45";
+            function = "gpio";
+            bias-pull-up;
+        };
+
+        host_int {
+            pins = "gpio42";
+            function = "gpio";
+            input;
+        };
+
+        mux {
+            pins = "gpio44", "gpio46", "gpio47";
+            function = "blsp_spi1";
+            bias-disable;
+        };
+
+        reset {
+            pins = "gpio49";
+            function = "gpio";
+            output-high;
+        };
+
+        wake {
+            pins = "gpio31";
+            function = "gpio";
+            output-high;
+        };
+    };
+
+    wifi_0_pins: wifi0_pinmux {
+        mux {
+            bias-pull-up;
+            drive-strength = <6>;
+            function = "gpio";
+            output-high;
+            pins = "gpio52";
+        };
+    };
+
+    
+    zigbee_0 {
+        gpio-hog;
+        gpios = <29 GPIO_ACTIVE_HIGH>;
+        bias-disable;
+        output-low;
+    };
+
+    zigbee_1 {
+        gpio-hog;
+        gpios = <50 GPIO_ACTIVE_HIGH>;
+        bias-disable;
+        input;
+    };
+
+    bluetooth_enable {
+        gpio-hog;
+        gpios = <32 GPIO_ACTIVE_HIGH>;
+        output-high;
+    };
+};
+
+&mdio {
+    status = "okay";
+    pinctrl-0 = <&mdio_pins>;
+    pinctrl-names = "default";
+    phy-reset-gpio = <&tlmm 19 GPIO_ACTIVE_LOW>;
+
+    ethphy0: ethernet-phy@0 {
+        status = "disabled";
+        reg = <0x18>;
+        #reg = <0x00>;
+    };
+
+    ethphy1: ethernet-phy@1 {
+        status = "disabled";
+        reg = <0x19>;
+        #reg = <0x01>;
+    };
+
+    ethphy2: ethernet-phy@2 {
+        status = "disabled";
+        reg = <0x1a>;
+        #reg = <0x02>;
+    };
+
+    ethphy3: ethernet-phy@3 {
+        status = "okay";
+        reg = <0x1b>;
+        #reg = <0x03>;
+    };
+
+    ethphy4: ethernet-phy@4 {
+        status = "okay";
+        reg = <0x1c>;
+        #reg = <0x04>;
+    };
+
+    psgmiiphy: psgmii-phy@5 {
+        reg = <0x1d>;
+        #reg = <0x05>;
+     };
+};
+
+&watchdog {
+    status = "okay";
+};
+
+&prng {
+    status = "okay";
+};
+
+&blsp_dma {
+    status = "okay";
+};
+
+&cryptobam {
+    num-channels = <4>;
+    qcom,num-ees = <2>;
```

Comment:
Do these properties even exist upstream?

Reply:
Yes, they are defined [here](https://github.com/torvalds/linux/blob/8bb7eca972ad531c9b149c0a51ab43a417385813/Documentation/devicetree/bindings/dma/qcom_bam_dma.txt#L18). 
I have added them after getting error messages during the boot. After some research it turned out that those properties are needed when the bam is remotely controlled.